### DISCUSSION TITLE
Lists not showing in admin from external domain when using IIS as a reverse proxy

### DISCUSSION THREAD

**Discussion by etshei** [strapi/strapi#7862] [Legacy Topics]
Hi,

We are working on our first project with strapi, and we are using **IIS** as a reverse proxy
Strapi is configured to use a subdomain like _api.domain.com_ and it's configured in the _config/server.js_ with `url: 'http://api.domain.com'`

Getting and creating/updating the objects are working as expected using the subdomain API endpoints but when accessing the list of the objects in the admin panel from _http://api.domain.com/admin_ there is an error saying "An error occured durring models config fetch." (when accessing from local everything is fine)

In the network pannel it's showing a **400 Bad Request** error when calling this endpoint (categorie model as an example) http://api.domain.com/content-manager/explorer/application::categories.categories/count?_limit=10&_sort=id%3AASC&_start=0

Creating and editing models in the admin panel works, showing the list of admin users from _setting/users_ works also. Listing the normal users have the same problem. 

Does anyone have any idea how to resolve this problem? 

Thanks in advance

PS: I think it's the same issue as in here #6672

**Comment by derrickmehaffy** [comment]
I'm not largely familiar with reverse proxying on IIS so I don't believe I will be able to provide much help to you, but if it is possible I would recommend to switch over to an application has better proxying support.

Specifically something like HAProxy which is built for it, or something like Nginx which has been proven to have much better performance.

**Comment by etshei** [comment]
For those having this issue, i was able to resolve it by following these steps (**the number 6 is the most important step!!**):

1- Installing **[iisnode](https://github.com/tjanczuk/iisnode)**
2- Pointing the iis website to the folder where the app is installed
3- Configuring your **web.config** by adding these lines in the **<system.webServer>** bloc:
```
<handlers>
        <add name="iisnode" path="server.js" verb="*" modules="iisnode" />
</handlers>
```
and this rule into the `<system.webServer> -> <rewrite> -> <rules>` bloc
```
<rule name="Strapi">
   <match url="/*" />
   <conditions logicalGrouping="MatchAll" trackAllCaptures="false" />
   <action type="Rewrite" url="server.js" />
</rule>
```
4- Creating a file named **server.js** in the app root with these two lines (as in [here](https://strapi.io/documentation/v3.x/getting-started/deployment.html#_2-launch-the-server)):
```
const strapi = require('strapi');
strapi().start();
```
5- Create a file named **iisnode.yml** in the app root 
```
node_env: development _(or production)_
watchedFiles: *.js;iisnode.yml;web.config
loggingEnabled: true  
logDirectory: iisnode 
debuggingEnabled: true 
```
6- in **config/server.js** replace the line setting the port from:
`port: env.int('PORT', 1337)`
to
`port: env('PORT', 1337)`

**In iisnode, the process.env.PORT is a string that represents the pipe and not an integer**

That's all folks

**Comment by etshei** [comment]
**server.js** in the previous reply (4th step) doesn't let you edit the models in the content-types-builder in development mode (it's like using **strapi start** _not_ **strapi develop**). 
For it to work, i've updated the file to be like this instead
```
if(process.env.NODE_ENV == 'development')
    strapi({autoReload: true}).start();
else
    strapi().start();
```

**Comment by derrickmehaffy** [comment]
@etshei If you are interested could you make a PR similar to those already existing ones for proxying with IIS to the docs.

Sample for the others: https://strapi.io/documentation/v3.x/getting-started/deployment.html#optional-software-guides
